### PR TITLE
Heatexchanger Cleanup and Dynamic NTU Condenser Init

### PR DESCRIPTION
**Pull request by eslickj**
**NOTE:** When this PR is approved, you *must* do the following so it can be merged:
* Add the "approved" tag to the pull-request
* Trigger tests by pushing a commit or manually re-running in CircleCI

See [this wiki page](https://github.com/IDAES/idaes-dev/wiki/Github-Pull-Request-Workflow-with-Approval-Tag) for details.

----

## Summary/Motivation:
This fixes an issue with dynamic initialization of the NTU Condenser power model. Does some minor rearrangement of the heat exchanger model to improve clarity regarding side names and adds tests for both the condenser and generic heat exchanger.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

----

## Reviewer Checklist
* Reviewer 1
- [ ] Tests: do they pass, do they cover functionality, are they understandable?
- [ ] Documentation: is the code commented, are there user-level docs where needed?
* Reviewer 2
- [ ] Tests: do they pass, do they cover functionality, are they understandable?
- [ ] Documentation: is the code commented, are there user-level docs where needed?

### CODE REVIEW

File: idaes/generic_models/unit_models/heat_exchanger.py
Comment (jghouse88) [inline idaes/generic_models/unit_models/heat_exchanger.py:167]: Suggest explicit naming like `temp_units` or `temperature_units`?

File: idaes/generic_models/unit_models/heat_exchanger.py
Comment (jghouse88) [inline idaes/generic_models/unit_models/heat_exchanger.py:314]: Can we avoid the alias for `self.heat_duty`? Might be easier to read the code instead of just `q`.

File: idaes/power_generation/unit_models/helm/condenser_ntu.py
Comment (jghouse88) [inline idaes/power_generation/unit_models/helm/condenser_ntu.py:236]: Should this be using `hot_side` and `cold_side` instead of `side_1` and `side_2`?

File: idaes/power_generation/unit_models/helm/condenser_ntu.py
Comment (jghouse88) [inline idaes/power_generation/unit_models/helm/condenser_ntu.py:371]: Just a general comment that `hot_side`/`cold_side` and `side_1`/`side_2` are used interchangeably. Not sure if this is intended.

**Review by jghouse88 — CHANGES_REQUESTED** [review]
Minor comments/questions. Looks good.

File: idaes/power_generation/unit_models/helm/condenser_ntu.py
Comment (eslickj) [reply idaes/power_generation/unit_models/helm/condenser_ntu.py:236]: Yes, although it doesn't matter.  Will fix.

File: idaes/generic_models/unit_models/heat_exchanger.py
Comment (eslickj) [reply idaes/generic_models/unit_models/heat_exchanger.py:167]: Sure.  Sounds good.  Will fix.

File: idaes/generic_models/unit_models/heat_exchanger.py
Comment (eslickj) [reply idaes/generic_models/unit_models/heat_exchanger.py:314]: Agree, will fix.

**Comment by eslickj** [comment]
Okay.  I addressed comments.  I also added a check to make sure someone doesn't name the hot side cold_side or side_2 and doesn't name the cold side hot_side or side_1.

**Review by jghouse88 — DISMISSED** [review]
LGTM. Probably better to wait for PR #54  to be merged before merging this..

File: idaes/generic_models/unit_models/heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/heat_exchanger.py:172]: The output of this should probably have units too; i.e. `units=pyunits.K`. This would mean you don't need to add units on line 179.

File: idaes/generic_models/unit_models/heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/heat_exchanger.py:223]: Minor point: `{'hot_side', 'side_1'}` should be a list.

File: idaes/generic_models/unit_models/heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/heat_exchanger.py:225]: As above.

File: idaes/generic_models/unit_models/heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/heat_exchanger.py:260]: Type: references

File: idaes/generic_models/unit_models/heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/heat_exchanger.py:261]: Typo: refer

File: idaes/generic_models/unit_models/heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/heat_exchanger.py:283]: I don't want to add more work, but it seems odd that this is `temp_units` whilst all the others are shorter now.

File: idaes/power_generation/unit_models/helm/condenser_ntu.py
Comment (andrewlee94) [inline idaes/power_generation/unit_models/helm/condenser_ntu.py:114]: As before: `{'hot_side', 'side_1'}` should be a list.

File: idaes/power_generation/unit_models/helm/condenser_ntu.py
Comment (andrewlee94) [inline idaes/power_generation/unit_models/helm/condenser_ntu.py:116]: As above.

File: idaes/power_generation/unit_models/helm/tests/test_condenser_ntu.py
Comment (andrewlee94) [inline idaes/power_generation/unit_models/helm/tests/test_condenser_ntu.py:27]: I would like to see the tests follow the standard of the core library more, but I won't insist on that.

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
A couple of minor comments, which are mostly formatting and typos.

File: idaes/generic_models/unit_models/heat_exchanger.py
Comment (eslickj) [reply idaes/generic_models/unit_models/heat_exchanger.py:283]: This was changed to address @jghouse88's comment.  I think it's just longer due to potential confusion between time and temperature.

File: idaes/power_generation/unit_models/helm/tests/test_condenser_ntu.py
Comment (eslickj) [reply idaes/power_generation/unit_models/helm/tests/test_condenser_ntu.py:27]: I'm a bit short on time right now, but hopefully very soon I'm going to update the tests and documentation for all the steam-related power models.

File: idaes/generic_models/unit_models/heat_exchanger.py
Comment (eslickj) [reply idaes/generic_models/unit_models/heat_exchanger.py:172]: I think this is ok.  I just left the cube root function return value unitless.  I figured this would make things simpler than having units of cube root of K the get cubed back to K later.  That process made me a little uncomfortable. Like what if I ended up with units of K^0.9999999999999 or something?

**Comment by eslickj** [comment]
@andrewlee94 and @jghouse88, I addressed @andrewlee94's comment and it's ready for another look.

File: idaes/generic_models/unit_models/heat_exchanger.py
Comment (eslickj) [reply idaes/generic_models/unit_models/heat_exchanger.py:172]: @jsiirola or @carldlaird, if we have units like (K^(1/3))^3 are we sure that the resulting unit is exactly K and not just close to K?

**Review by jghouse88 — APPROVED** [review]
LGTM. I am seeing that it is out of date with the main branch.

**Review by andrewlee94 — APPROVED** [review]
Looks good. I would like to resolve the concern about units in the cube root function, but that is not critical.

**Pull request merged**
